src/parser/Parser.ts

```diff
@@ -1449,12 +1449,12 @@ export class Parser {
         return annotation;
     }
 
-    private ternaryExpression(test?: Expression): TernaryExpression {
-        this.warnIfNotBrighterScriptMode('ternary operator');
+    private ternaryOrIndexedGet(test?: Expression): TernaryExpression | IndexedGetExpression {
         if (!test) {
             test = this.expression();
         }
         const questionMarkToken = this.advance();
+        const start = this.current;
 
         //consume newlines or comments
         while (this.checkAny(TokenKind.Newline, TokenKind.Comment)) {
```

Comment:
Should that immediately bust out the possibility of an indexed get?

Reply:
Yes, we can assume it won't be an indexedGet if we see any token other than a left square brace. However, this function parses ternary by default and then scraps ternary in favor of optional chained indexedGet. So, there's really nothing to bust out of here.